tests/trainer/logging_/test_eval_loop_logging.py

```diff
@@ -929,3 +930,57 @@ def test_rich_print_results(inputs, expected):
     EvaluationLoop._print_results(*inputs, file=out)
     expected = expected[1:]  # remove the initial line break from the """ string
     assert out.getvalue() == expected.lstrip()
+
+
+@pytest.mark.parametrize("num_dataloaders", [1, 2])
+def test_eval_step_logging(tmpdir, num_dataloaders):
+    class CustomLogger(CSVLogger):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.logged_metrics = collections.defaultdict(lambda: collections.defaultdict(list))
+
+        def log_metrics(self, metrics, step):
+            for key in metrics:
+                self.logged_metrics[key]["logged_values"].append(metrics[key])
+                self.logged_metrics[key]["logged_steps"].append(step)
```

Comment:
Using the CSV logger to test this is a bit ugly.

Why not mock `logger.log_metrics` instead and test the call inputs?

Reply:
can be done that way too. will update